File: projeto/.metadata/climedic/src/main/java/br/com/puc/tcc/service/ResultadoExameService.java
package br.com.puc.tcc.service;

import java.util.List;

import br.com.puc.tcc.dominio.ResultadoExame;
import br.com.puc.tcc.dominio.retorno.ResultadoExameRetorno;

public interface ResultadoExameService {
	
	void salvarResultadoExame(ResultadoExame resultadoExame);
	
	void editarResultadoExame(ResultadoExame resultadoExame);
	
	void deletarResultadoExame(Integer idConsulta);
	
	List<ResultadoExameRetorno> getResultadoExames();
	
	ResultadoExameRetorno getResultadoExame(Integer idConsulta);
	
	List<ResultadoExameRetorno> getResultadoExamesPorProfissional(Integer idProfissional);
}

File: projeto/climed-app/src/app/services/resultado-exame/resultado-exame.service.ts
import { Injectable } from '@angular/core';
import { Http} from "@angular/http"
import { Headers} from '@angular/http';
import { RequestOptions } from '@angular/http';
 
import 'rxjs/add/operator/map';
import 'rxjs/add/operator/do';
import 'rxjs/add/operator/catch';
 
import {ResultadoExame} from './resultado-exame';
import {ConfigService} from '../config.service';
 
@Injectable()
export class ResultadoExameService {
 
    private baseUrlService:string = '';
    private headers:Headers;
    private options:RequestOptions;
 
    constructor(private http: Http, private configService: ConfigService) { 
        this.baseUrlService = configService.getUrlService();
        this.headers = new Headers({ 'Content-Type': 'application/json;charset=UTF-8' });                
        this.options = new RequestOptions({ headers: this.headers });
    }

    public getResultadoExames(){        
        return this.http.get(this.baseUrlService + "/resultadoexames/todos/").map(res => res.json());
   }
    
   public getResultadoExamesPorProfissional(idProfissional:number){        
    return this.http.get(this.baseUrlService + "/resultadoexames/profissional/" + idProfissional).map(res => res.json());
   }

    getResultadoExame(idConsulta:number){
        return this.http.get(this.baseUrlService + "/resultadoexame/dados/" + idConsulta).map(res => res.json());
    }

    addResultadoExame(re: ResultadoExame){
        return this.http.post(this.baseUrlService + "/resultadoexame/novo", JSON.stringify(re),this.options).map(res => res.json());
    }
 
    atualizarResultadoExame(re:ResultadoExame){
        return this.http.post(this.baseUrlService + "/resultadoexame/atualizar" , JSON.stringify(re),this.options).map(res => res.json());
    }

    excluirResultadoExame(idConsulta:number){
        return this.http.delete(this.baseUrlService + "/resultadoexame/deletar/" + idConsulta).map(res => res.json());
    }

}
